hadoop-tools/hadoop-aws/src/test/java/org/apache/hadoop/fs/s3a/ITestS3GuardOutOfBandOperations.java

```diff
@@ -271,6 +281,202 @@ public void testListingDelete() throws Exception {
     deleteFileInListing();
   }
 
+  /**
+   * Tests that tombstone expiry is implemented, so if a file is created raw
+   * while the tombstone exist in ms for with the same name then S3Guard will
+   * check S3 for the file.
+   *
+   * Seq: create guarded; delete guarded; create raw (same path); read guarded;
+   * This will fail if no tombstone expiry is set
+   *
+   * @throws Exception
+   */
+  @Test
+  public void testTombstoneExpiryGuardedDeleteRawCreate() throws Exception {
```

Comment:
I want to propose a new test, and it's going to have to be in its own test case.

1. File is deleted in the guarded fs.
1. File is replaced in the raw fs.
1. File is deleted in the guarded FS after the expiry time.
1. File MUST NOT exist in raw FS.

that is: do delete() operations also use the state. I'm assuming they will, but let's make sure.